toolchain/lower/file_context.h

```diff
@@ -219,6 +294,34 @@ class FileContext {
 
   // Global format string for `printf.int.format` used by the PrintInt builtin.
   llvm::Value* printf_int_format_string_ = nullptr;
+
+  // For a generic function, keep track of the specifics for which LLVM
+  // function declarations were created. Those can be retrieved then via
+  // from specific_functions_, via specific_functions_[specific_id.index].
+  // We resize this to the correct size. Indexed by generic_id.index.
+  llvm::SmallVector<llvm::SmallVector<SemIR::SpecificId>, 0> lowered_specifics_;
+
+  // For specifics that exist in lowered_specifics, a hash of their function
+  // type information: return and parameter types. We resize this to the
+  // correct size. Indexed by specific_id.index.
+  // TODO: Hashing all members of FunctionTypeInfo may not be necessary.
+  llvm::SmallVector<llvm::BLAKE3Result<32UL>, 0>
+      lowered_specifics_type_fingerprint_;
+
+  // This is initialized and populated while lowering a specific.
+  // We resize this to the correct size. Indexed by specific_id.index.
+  llvm::SmallVector<SpecificFunctionFingerprint, 0>
+      lowered_specific_fingerprint_;
+
+  // Equivalent specifics found: for each specific point to the
+  // canonical equivalent specific
+  llvm::SmallVector<SemIR::SpecificId> equivalent_specifics_;
+  // Non-equivalent specifics found.
+  Set<std::pair<SemIR::SpecificId, SemIR::SpecificId>>
```

Comment:
I wonder if we should be considering how this grows, particularly when there are many specifics for the same type. Do you think that's something we should have a TODO for?